apps/wallet/src/ui/app/components/accounts/AccountBalanceItem.tsx

```diff
@@ -1,46 +1,177 @@
+        return `${formatted} ${symbol}`;
+    }
+
+    function getSumOfBalances(): string {
+        let coinType = '';
+        const balance = accounts.reduce((acc, { address }) => {
+            const { data: balance } = useBalance(address, {
+                refetchInterval: false,
+            });
+            const totalBalance = balance?.totalBalance || '0';
+            coinType = balance?.coinType || '';
+            return (BigInt(acc) + BigInt(totalBalance)).toString();
+        }, '0');
+        const [formatted, symbol] = useFormatCoin(BigInt(balance), coinType);
+        return `${formatted} ${symbol}`;
+    }
+
+    function hasAccountAssets(): boolean {
+        return accounts.some(({ address }) => {
+            const { data: ownedAssets } = useGetOwnedObjects(
+                address,
+                {
+                    MatchNone: [
+                        { StructType: COIN_TYPE },
+                        { StructType: TIMELOCK_IOTA_TYPE },
+                        { StructType: TIMELOCK_STAKED_TYPE },
+                        { StructType: STARDUST_BASIC_OUTPUT_TYPE },
+                        { StructType: STARDUST_NFT_OUTPUT_TYPE },
+                    ],
+                },
+                1,
+            );
+            return (ownedAssets && ownedAssets?.pages?.[0]?.data?.length > 0) ?? false;
+        });
+    }
+
+    function hasSupplyIncreaseVestingObjects(): boolean {
+        return accounts.some(({ address }) => {
+            const { data: supplyIncreaseVestingObjects } = useGetOwnedObjects(
+                address,
+                {
+                    MatchAny: [
+                        { StructType: TIMELOCK_IOTA_TYPE },
+                        { StructType: TIMELOCK_STAKED_TYPE },
+                    ],
+                },
+                1,
+            );
+            return (
+                (supplyIncreaseVestingObjects &&
+                    supplyIncreaseVestingObjects?.pages?.[0]?.data?.length > 0) ??
+                false
+            );
+        });
+    }
+
+    function hasMigrationObjects(): boolean {
+        const OBJECT_PER_REQ = 1;
+        let containsMigrationObjects = false;
+        return accounts.some(({ address }) => {
+            const { data: stardustSharedBasicObjects } = useGetStardustSharedBasicObjects(
+                address || '',
+                OBJECT_PER_REQ,
+            );
+            const { data: stardustSharedNftObjects } = useGetStardustSharedNftObjects(
+                address || '',
+                OBJECT_PER_REQ,
+            );
+
+            const { data: legacyObjects } = useGetOwnedObjects(
+                address,
+                {
+                    MatchAny: [
+                        { StructType: STARDUST_BASIC_OUTPUT_TYPE },
+                        { StructType: STARDUST_NFT_OUTPUT_TYPE },
+                    ],
+                },
+                1,
+            );
```

Comment:
we could make these requests one by one, so that if there is an `stardustSharedBasicObjects`, it is not necessary to make the following 2 requests

Reply:
hooks can't be inside of conditions https://react.dev/reference/rules/rules-of-hooks#only-call-hooks-at-the-top-level :pray: